### PR TITLE
Fix Decimal for non-standard integers

### PR DESCRIPTION
**Pull request by wismill**
- Allow Int30 to have negative values.
- Fix 2 errors in `Decimal`’s code.

### CODE REVIEW

File: src/Data/Text/Builder/Linear/Dec.hs [hunk 1]
```diff
@@ -108,8 +108,12 @@ unsafePrependDec marr !off n
     digits = "00010203040506070809101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899"#
 {-# INLINEABLE unsafePrependDec #-}
 
+-- minBound = 1 `shiftL` (finiteBitSize a - 1) = -2^(finiteBitSize a - 1)
+-- Note that the last digit of 2^k forms a cycle for k≥ 1: 2,4,8,6
+-- Thus it is enough to compute rem (finiteBitSize a) 4, i.e. finiteBitSize a .&. 3
+-- This avoid the need for the more expansive rem n 10.
```
Comment (Bodigrim) [inline src/Data/Text/Builder/Linear/Dec.hs:114]:
```suggestion
-- This avoid the need for the more expensive rem n 10.
```

File: src/Data/Text/Builder/Linear/Dec.hs [hunk 2]
```diff
@@ -108,8 +108,12 @@ unsafePrependDec marr !off n
...
 minBoundLastDigit ∷ FiniteBits a ⇒ a → Int
-minBoundLastDigit a = case finiteBitSize a .&. 4 of
+minBoundLastDigit a = case finiteBitSize a .&. 3 of
```
Comment (Bodigrim) [inline src/Data/Text/Builder/Linear/Dec.hs:116]: Thanks, that's embarassing :)

Could you possibly check with `--enable-coverage` that we now exercise all alternatives below?

File: src/Data/Text/Builder/Linear/Dec.hs (hunk 1 above)
Comment (wismill) [reply src/Data/Text/Builder/Linear/Dec.hs]: Good catch. I ended reworking the complete comment.

File: src/Data/Text/Builder/Linear/Dec.hs (hunk 2 above)
Comment (wismill) [reply src/Data/Text/Builder/Linear/Dec.hs:117]: Good idea to check coverage!

I refactored `Int30` to the more general `IntN n` to test the following integer sizes: 8, 16, 30, 31, 32, 33 , 64. We now have 100% coverage for `Data.Text.Builder.Linear.Dec`. Note: I need to expose it temporarily in order to check this.

**Comment by Bodigrim** [comment]
Looks great, thank you!

**Pull request merged**